api/cas/views.py

```diff
@@ -0,0 +1,310 @@
+import json
+
+from rest_framework import generics
+from rest_framework import status
+from rest_framework.exceptions import ValidationError
+from rest_framework.response import Response
+
+from api.base.views import JSONAPIBaseView
+from api.cas import login, account, messages, permissions, service
+from api.cas.mixins import APICASMixin
+
+
+class LoginOsf(APICASMixin, generics.CreateAPIView, JSONAPIBaseView):
+    """ Default login through OSF. Authentication Required.
+    """
+
+    view_category = 'cas'
+    view_name = 'login-osf'
+    permission_classes = (permissions.IsCasLogin,)
+    authentication_classes = (login.CasLoginAuthentication,)
+
+    def post(self, request, *args, **kwargs):
+
+        content = {
+            'userId': request.user._id,
+            'attributes': {
+                'username': request.user.username,
+                'givenName': request.user.given_name,
+                'familyName': request.user.family_name,
+            }
+        }
+
+        return Response(data=content, status=status.HTTP_200_OK)
+
+
+class LoginInstitution(APICASMixin, generics.CreateAPIView, JSONAPIBaseView):
+    """ Login through institutions. Authentication Required.
+    """
+
+    view_category = 'cas'
+    view_name = 'login-institution'
+    permission_classes = (permissions.IsCasLogin,)
+    authentication_classes = (login.CasLoginAuthentication,)
+
+    def post(self, request, *args, **kwargs):
+
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class LoginExternal(APICASMixin, generics.CreateAPIView, JSONAPIBaseView):
```

Comment:
External login view. Everything is handled into the `CasLoginAuthentication` class.